src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -440,52 +439,58 @@ void Instance::NotifyProgressChanged()
-    // If the LandmarkTag field is null, the HomeLocationInfo field SHALL NOT be null.
-    if (aLocation.locationInfo.locationInfo.IsNull() && aLocation.locationInfo.landmarkTag.IsNull())
+    // If the LocationInfo field is null, the LandmarkInfo field SHALL NOT be null.
+    // If the LandmarkInfo field is null, the LocationInfo field SHALL NOT be null.
+    if (aLocation.areaDesc.locationInfo.IsNull() && aLocation.areaDesc.landmarkInfo.IsNull())
     {
-        ChipLogDetail(Zcl, "IsValidAsSupportedLocation %u - must have locationInfo and/or LandmarkTag", aLocation.locationID);
+        ChipLogDetail(Zcl, "IsValidAsSupportedLocation %u - must have locationInfo and/or LandmarkInfo", aLocation.areaID);
         return false;
     }
 
-    // If HomeLocationInfo is not null, and its LocationName field is an empty string, at least one of the following SHALL NOT
-    // be null: HomeLocationInfo's FloorNumber field, HomeLocationInfo's AreaType field, the LandmarkTag field
-    if (!aLocation.locationInfo.locationInfo.IsNull())
+    // If LocationInfo is not null, and its LocationName field is an empty string, at least one of the following SHALL NOT
+    // be null: LocationInfo's FloorNumber field, LocationInfo's AreaType field, the LandmarkInfo
+    if (!aLocation.areaDesc.locationInfo.IsNull())
     {
-        if (aLocation.locationInfo.locationInfo.Value().locationName.empty() &&
-            aLocation.locationInfo.locationInfo.Value().floorNumber.IsNull() &&
-            aLocation.locationInfo.locationInfo.Value().areaType.IsNull() && aLocation.locationInfo.landmarkTag.IsNull())
+        if (aLocation.areaDesc.locationInfo.Value().locationName.empty() &&
+            aLocation.areaDesc.locationInfo.Value().floorNumber.IsNull() &&
+            aLocation.areaDesc.locationInfo.Value().areaType.IsNull() && aLocation.areaDesc.landmarkInfo.IsNull())
         {
             ChipLogDetail(
-                Zcl, "IsValidAsSupportedLocation %u - LocationName is empty string, FloorNumber, AreaType, LandmarkTag are null",
-                aLocation.locationID);
+                Zcl, "IsValidAsSupportedLocation %u - LocationName is empty string, FloorNumber, AreaType, LandmarkInfo are null",
+                aLocation.areaID);
             return false;
         }
     }
 
-    // If the LandmarkTag field is null, the PositionTag field SHALL be null.
-    if (aLocation.locationInfo.landmarkTag.IsNull() && !aLocation.locationInfo.positionTag.IsNull())
+    // The mapID field SHALL be null if the SupportedMaps is not supported on the SupportedMaps is an empty list.
+    bool shouldMapsBeNull = false;
+    if (mFeature.Has(Feature::kMaps))
     {
-        ChipLogDetail(Zcl, "IsValidAsSupportedLocation %u - PositionTag with no LandmarkTag", aLocation.locationID);
-        return false;
+        if (mDelegate->GetNumberOfSupportedMaps() == 0)
+        {
+            shouldMapsBeNull = true;
+        }
+    }
+    else
+    {
+        shouldMapsBeNull = true;
```

Comment:
(nit) lines 477-487 - minor suggestion for a more compact form:

shouldMapsBeNull = ! mFeature.Has(Feature::kMaps) ||  (mDelegate->GetNumberOfSupportedMaps() == 0)

Reply:
If `mFeature.Has(Feature::kMaps)` in false then `mDelegate->GetNumberOfSupportedMaps()` may return an error, such as "not implemented". In this case we do not want this to fail.